packages/playwright-core/src/server/recorder/python.ts

```diff
@@ -115,35 +115,37 @@ export class PythonLanguageGenerator implements LanguageGenerator {
       case 'uncheck':
-        return `${subject}.${this._asLocator(action.selector)}.uncheck()`;
+        return `${this._awaitPrefix}${subject}.${this._asLocator(action.selector)}.uncheck()`;
       case 'fill':
-        return `${subject}.${this._asLocator(action.selector)}.fill(${quote(action.text)})`;
+        return `${this._awaitPrefix}${subject}.${this._asLocator(action.selector)}.fill(${quote(action.text)})`;
       case 'setInputFiles':
-        return `${subject}.${this._asLocator(action.selector)}.set_input_files(${formatValue(action.files.length === 1 ? action.files[0] : action.files)})`;
+        return `${this._awaitPrefix}${subject}.${this._asLocator(action.selector)}.set_input_files(${formatValue(action.files.length === 1 ? action.files[0] : action.files)})`;
       case 'press': {
         const modifiers = toModifiers(action.modifiers);
         const shortcut = [...modifiers, action.key].join('+');
-        return `${subject}.${this._asLocator(action.selector)}.press(${quote(shortcut)})`;
+        return `${this._awaitPrefix}${subject}.${this._asLocator(action.selector)}.press(${quote(shortcut)})`;
       }
       case 'navigate':
-        return `${subject}.goto(${quote(action.url)})`;
+        return `${this._awaitPrefix}${subject}.goto(${quote(action.url)})`;
       case 'select':
-        return `${subject}.${this._asLocator(action.selector)}.select_option(${formatValue(action.options.length === 1 ? action.options[0] : action.options)})`;
+        return `${this._awaitPrefix}${subject}.${this._asLocator(action.selector)}.select_option(${formatValue(action.options.length === 1 ? action.options[0] : action.options)})`;
       case 'assertText':
-        return `expect(${subject}.${this._asLocator(action.selector)}).${action.substring ? 'to_contain_text' : 'to_have_text'}(${quote(action.text)})`;
+        return `${this._awaitPrefix}expect(${subject}.${this._asLocator(action.selector)}).${action.substring ? 'to_contain_text' : 'to_have_text'}(${quote(action.text)})`;
       case 'assertChecked':
-        return `expect(${subject}.${this._asLocator(action.selector)}).${action.checked ? 'to_be_checked()' : 'not_to_be_checked()'}`;
+        return `${this._awaitPrefix}expect(${subject}.${this._asLocator(action.selector)}).${action.checked ? 'to_be_checked()' : 'not_to_be_checked()'}`;
       case 'assertVisible':
-        return `expect(${subject}.${this._asLocator(action.selector)}).to_be_visible()`;
+        return `${this._awaitPrefix}expect(${subject}.${this._asLocator(action.selector)}).to_be_visible()`;
       case 'assertValue': {
         const assertion = action.value ? `to_have_value(${quote(action.value)})` : `to_be_empty()`;
-        return `expect(${subject}.${this._asLocator(action.selector)}).${assertion};`;
+        return `${this._awaitPrefix}expect(${subject}.${this._asLocator(action.selector)}).${assertion}`;
       }
+      case 'assertScreenshot':
```

Comment:
maybe its better to disable the assertion functionality for the language bindings?